### PR TITLE
D-17804 (post D-17750) History corruption

### PR DESCRIPTION
**Pull request by lremedi**
Make sure the current record is only a description change by the current user.

### CODE REVIEW

File: fixes/consolidate-multiple-description-history.sql [hunk 1]
```diff
@@ -56,6 +56,7 @@ select _.ID, CurrentAuditID, NextAuditID
 from #suspect _
 join dbo.BaseAsset A on A.ID=_.ID and A.AuditBegin=_.CurrentAuditID
 join dbo.BaseAsset B on B.ID=_.ID and B.AuditBegin=_.NextAuditID
+join dbo.BaseAsset C on C.ID=_.ID and C.AuditEnd=_.CurrentAuditID AND ISNULL(C.[Description],-1) != ISNULL(A.[Description],-1)
```
Comment (spazmodius) [inline fixes/consolidate-multiple-description-history.sql:59]: I'm not sure I understand. Is the idea to only pick up rows sandwiched between 2 Decription-only changes?

Comment (pmamut) [reply fixes/consolidate-multiple-description-history.sql:59]: Originally we picked "current" `BaseAsset` rows that have changes only to description, by the same user, within specified time interval, and without any other in-between changes ("next"). We then dropped "current", and re-stitched history,  expanding `BaseAsset` preceding the "current" to end at "next", effectively preserving only the most recent "next" change.

We now also ensure that "current" row actually introduced only a description change before dropping it (we already know it was made by the same user). This ensures "creation" rows don't get dropped (no preceding record), and "current" rows that weren't a description change (ie changes to name, asset state, or security scope) are also preserved.

Comment (spazmodius) [reply fixes/consolidate-multiple-description-history.sql:59]: So, why is it important for `B` to be a description-only change?

Comment (pmamut) [reply fixes/consolidate-multiple-description-history.sql:59]: Good point - it isn't important for `B` to be a description-only change.
It's the `A` that we should be enforcing to be a description-only change in order to target it for removal.

File: fixes/consolidate-multiple-description-history.sql [hunk 2]
```diff
@@ -56,7 +56,7 @@ select _.ID, CurrentAuditID, NextAuditID
 from #suspect _
 join dbo.BaseAsset A on A.ID=_.ID and A.AuditBegin=_.CurrentAuditID
 join dbo.BaseAsset B on B.ID=_.ID and B.AuditBegin=_.NextAuditID
-join dbo.BaseAsset C on C.ID=_.ID and C.AuditEnd=_.CurrentAuditID AND ISNULL(C.[Description],-1) != ISNULL(A.[Description],-1)
+join dbo.BaseAsset C on C.ID=_.ID and C.AuditEnd=_.CurrentAuditID
```
Comment (spazmodius) [inline fixes/consolidate-multiple-description-history.sql:59]: I'm confused.  What scenario are we targeting, exactly?

A Description-only change, followed quickly by a same-user Description-plus change?  If so, I don't think this commit does that.

**Pull request merged**